### PR TITLE
Remove unneeded requirements file

### PR DESCRIPTION
**Pull request by joselsegura**
# Description

Adding Tox and PBR to enforce the usage of requirements.txt in a better way.
Adding a constraints file too in order to create "reproducible dependencies".  In this case, only for testing

## Type of change

- Refactor (refactoring code, removing useless files)
- Bump-up dependent library (no changes in the code)

## Testing steps

Tested locally with tox.

## Checklist
* [ ] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by tisnik — APPROVED** [review]
LGTM

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[tox]
+minversion = 3.18.0
+envlist = py38, py39, py310, py311, py312
```
Comment (danielmellado) [inline tox.ini:3]: I'd just change this to py{38,39,310,311,312}

File: requirements-tests.txt
Comment (danielmellado) [inline requirements-tests.txt:1]: please change this name to `test-requirements.txt` to match the default style

File: requirements-tests.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+freezegun
```
Comment (danielmellado) [inline requirements-tests.txt:1]: overall, I'd totally set the versions for this, not doing so may be prone to errors when running tests. In doubt, check which ones are you currently using and add those.

File: requirements.txt [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
+app-common-python>=0.2.3
+boto3<=1.28.64
+confluent-kafka>=2.0.0
+insights-core>=3.1.2
+insights-core-messaging>=1.2.4
+jsonschema>=4.0.0
+prometheus-client>=0.16.0
+python-json-logger
+PyYAML>=6.0
+requests
```
Comment (danielmellado) [inline requirements.txt:10]: Same as in the other file, which version of `requests`?

File: tox.ini (hunk 1 above)
Comment (joselsegura) [reply tox.ini]: Thanks!

File: requirements-tests.txt (hunk 2 above)
Comment (joselsegura) [reply requirements-tests.txt]: I already pinpointed all the requirements from this file

File: requirements.txt (hunk 3 above)
Comment (joselsegura) [reply requirements.txt]: Done

File: requirements-tests.txt
Comment (joselsegura) [reply requirements-tests.txt:1]: Done

**Review by danielmellado — APPROVED** [review]
Thanks, I'd take a look at adding the linters to the tox configuration as well and a few more nits, but that can be done in a follow-up patch ;)

**Review by Bee-lee — APPROVED** [review]

**Pull request merged**